### PR TITLE
Fix 'this._meta.byKey is not a function' error

### PR DESCRIPTION
**Pull request by nikitindv**
Hi, in old browsers i am getting exception about this._meta.byKey is not a function.
This exception caused by not registered iron-meta component at moment of iron-icon initialization due to webcomponents polyfill register components in asynchronous order.
So we should use whenDefined webcomponents api to await for iron-meta registered

### CODE REVIEW

**Comment by nikitindv** [comment]
I signed it!

**Comment by iexplore** [comment]
I just upgraded my large app to Polymer 2 (using latest versions) and have this same problem on Firefox.
It is triggered by a Firefox plugin Lastpass although other websites are working fine. 
Apparently this line "...this._meta.byKey..." is not very stable (considering the earlier issues).

The fix of @nikitindv is working and solves the problem.

**Comment by JonathanSidego** [comment]
Any news on this?
It's quite a high-priority fix.

File: iron-icon.html [hunk 1]
```diff
@@ -107,108 +107,112 @@
...
-      _updateIcon: function() {
-        if (this._usesIconset()) {
-          if (this._img && this._img.parentNode) {
-            Polymer.dom(this.root).removeChild(this._img);
-          }
-          if (this._iconName === "") {
-            if (this._iconset) {
-              this._iconset.removeIcon(this);
+    (function (window) {
+      window.customElements.whenDefined('iron-meta').then(function () {
```
Comment (justinfagnani) [inline iron-icon.html:111]: Given the HTML import of iron-meta above, it should _always_ be defined by the time iron-icon is. There's some other bug here. Are we seeing this fix things only with the polyfill or when bundled? cc @usergenic @valdrinkoshi

Comment (usergenic) [reply iron-icon.html:111]: @nikitindv Firefox and browsers which require the custom elements polyfill are vulnerable to having the polyfills broken when extensions (such as LastPass) are installed which patch their own version of the "createElement" method onto "document".  There is a workaround for this particular scenario; can you provide reproduction instructions for the error?

**Comment by usergenic** [comment]
@nikitindv have a look at this workaround while LastPass is in process of fixing their extension.  The essential problem is a conflict over control for `document.createElement`.  More discussion here https://github.com/webcomponents/custom-elements/issues/137

**Comment by usergenic** [comment]
Link to workaround: https://gist.github.com/usergenic/9baf88a24dcbc77bc808339755a7d436

**Pull request closed without merging**